src/PrefixTree.java

```diff
@@ -0,0 +1,62 @@
+import java.util.ArrayList;
+import java.util.Optional;
+
+public class PrefixTree {
+
+    private final Node root;
+
+    PrefixTree() {
+        this.root = new Node('\u0000');
+    }
+
+    private static class TraverseResult {
+        private Node node;
+        private int index;
+
+    }
+
+    public boolean contains(CharSequence word) {
+        TraverseResult traverseResult = traverse(word);
+        if (traverseResult.index == word.length() && traverseResult.node.isFinal()) {
+            System.out.println("Tree contains this word - " + word);
```

Comment:
The `println` calls are all debug lines which won't stay so thay too should get a TODO (see #4) or simply be removed.